diff --git a/_upgrade_staging/scintilla557/scintilla/include/Sci_Position.h b/scintilla/include/Sci_Position.h
index abd0f3408..0ebdf5983 100644
--- a/_upgrade_staging/scintilla557/scintilla/include/Sci_Position.h
+++ b/scintilla/include/Sci_Position.h
@@ -9,7 +9,11 @@
 #ifndef SCI_POSITION_H
 #define SCI_POSITION_H
 
+#if defined(__cplusplus)
+#include <cstddef>
+#else
 #include <stddef.h>
+#endif
 
 // Basic signed type used throughout interface
 typedef ptrdiff_t Sci_Position;
