diff --git a/_upgrade_staging/scintilla557/scintilla/include/ILoader.h b/scintilla/include/ILoader.h
index 5d2eb2571..9d49c75e8 100644
--- a/_upgrade_staging/scintilla557/scintilla/include/ILoader.h
+++ b/scintilla/include/ILoader.h
@@ -18,7 +18,7 @@ public:
 	virtual int SCI_METHOD Release() = 0;
 	// Returns a status code from SC_STATUS_*
 	virtual int SCI_METHOD AddData(const char *data, Sci_Position length) = 0;
-	virtual void * SCI_METHOD ConvertToDocument() = 0;
+	virtual void * SCI_METHOD ConvertToDocument() noexcept = 0;
 };
 
 static constexpr int deRelease0 = 0;
