diff --git a/_upgrade_staging/scintilla557/scintilla/win32/PlatWin.h b/scintilla/win32/PlatWin.h
index 1edad6044..8ebf5bd7a 100644
--- a/_upgrade_staging/scintilla557/scintilla/win32/PlatWin.h
+++ b/scintilla/win32/PlatWin.h
@@ -18,6 +18,8 @@ constexpr FLOAT dpiDefault = USER_DEFAULT_SCREEN_DPI;
 // Used for defining font size with LOGFONT
 constexpr int pointsPerInch = 72;
 
+constexpr UINT codePageWindowsLatin = 1252;
+
 extern void Platform_Initialise(void *hInstance) noexcept;
 
 extern void Platform_Finalise(bool fromDllMain) noexcept;
@@ -68,7 +70,12 @@ float GetDeviceScaleFactorWhenGdiScalingActive(HWND hWnd) noexcept;
 
 int SystemMetricsForDpi(int nIndex, UINT dpi) noexcept;
 
-void AdjustWindowRectForDpi(LPRECT lpRect, DWORD dwStyle, UINT dpi) noexcept;
+// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+#ifndef AlphaBlend
+#define AlphaBlend  GdiAlphaBlend
+#endif
+BOOL AdjustWindowRectForDpi(LPRECT lpRect, DWORD dwStyle, DWORD dwExStyle, UINT dpi)  noexcept;
+// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 
 HCURSOR LoadReverseArrowCursor(UINT dpi) noexcept;
 
