diff --git a/_upgrade_staging/scintilla/src/Editor.h b/scintilla/src/Editor.h
index f8bc18970..26f3c1941 100644
--- a/_upgrade_staging/scintilla/src/Editor.h
+++ b/scintilla/src/Editor.h
@@ -69,13 +69,17 @@ class SelectionText {
 public:
 	bool rectangular;
 	bool lineCopy;
+	// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+	bool asBinary;
 	int codePage;
 	Scintilla::CharacterSet characterSet;
-	SelectionText() noexcept : rectangular(false), lineCopy(false), codePage(0), characterSet(Scintilla::CharacterSet::Ansi) {}
+	SelectionText() noexcept : rectangular(false), lineCopy(false), asBinary(false), codePage(0), characterSet(Scintilla::CharacterSet::Ansi) {}
 	void Clear() noexcept {
 		s.clear();
 		rectangular = false;
 		lineCopy = false;
+		asBinary = false;
+	// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 		codePage = 0;
 		characterSet = Scintilla::CharacterSet::Ansi;
 	}
@@ -462,7 +466,7 @@ protected:	// ScintillaBase subclass needs access to much of Editor
 	virtual void NotifyChange() = 0;
 	virtual void NotifyFocus(bool focus);
 	virtual void SetCtrlID(int identifier);
-	virtual int GetCtrlID() { return ctrlID; }
+	virtual int GetCtrlID() const noexcept { return ctrlID; }
 	virtual void NotifyParent(Scintilla::NotificationData scn) = 0;
 	virtual void NotifyStyleToNeeded(Sci::Position endStyleNeeded);
 	void NotifyChar(int ch, Scintilla::CharacterSource charSource);
@@ -537,7 +541,7 @@ protected:	// ScintillaBase subclass needs access to much of Editor
 	void CopyText(size_t length, const char *text);
 	void SetDragPosition(SelectionPosition newPos);
 	virtual void DisplayCursor(Window::Cursor c);
-	virtual bool DragThreshold(Point ptStart, Point ptNow);
+	virtual bool DragThreshold(Point ptStart, Point ptNow) noexcept;
 	virtual void StartDrag();
 	void DropAt(SelectionPosition position, const char *value, size_t lengthValue, bool moving, bool rectangular);
 	void DropAt(SelectionPosition position, const char *value, bool moving, bool rectangular);
@@ -583,9 +587,9 @@ protected:	// ScintillaBase subclass needs access to much of Editor
 	virtual void IdleWork();
 	virtual void QueueIdleWork(WorkItems items, Sci::Position upTo=0);
 
-	virtual int SupportsFeature(Scintilla::Supports feature);
-	virtual bool PaintContains(PRectangle rc);
-	bool PaintContainsMargin();
+	virtual int SupportsFeature(Scintilla::Supports feature) const noexcept;
+	virtual bool PaintContains(PRectangle rc) const noexcept;
+	bool PaintContainsMargin() const noexcept;
 	void CheckForChangeOutsidePaint(Range r);
 	void SetBraceHighlight(Sci::Position pos0, Sci::Position pos1, int matchStyle);
 
