diff --git a/_upgrade_staging/scintilla/src/Style.cxx b/scintilla/src/Style.cxx
index 2c78c5241..16643fb95 100644
--- a/_upgrade_staging/scintilla/src/Style.cxx
+++ b/scintilla/src/Style.cxx
@@ -74,6 +74,9 @@ Style::Style(const char *fontName_) noexcept :
 	back(white),
 	eolFilled(false),
 	underline(false),
+	// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+	strike(false),
+	// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 	caseForce(CaseForce::mixed),
 	visible(true),
 	changeable(true),
