diff --git a/_upgrade_staging/scintilla/src/CellBuffer.h b/scintilla/src/CellBuffer.h
index 1b035597a..23916abc4 100644
--- a/_upgrade_staging/scintilla/src/CellBuffer.h
+++ b/scintilla/src/CellBuffer.h
@@ -8,6 +8,10 @@
 #ifndef CELLBUFFER_H
 #define CELLBUFFER_H
 
+// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+#include "Position.h"
+// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
+
 namespace Scintilla::Internal {
 
 // Interface to per-line data that wants to see each line insertion and deletion
