diff --git a/_upgrade_staging/scintilla/src/ViewStyle.h b/scintilla/src/ViewStyle.h
index 6985e3f03..28474d536 100644
--- a/_upgrade_staging/scintilla/src/ViewStyle.h
+++ b/scintilla/src/ViewStyle.h
@@ -8,6 +8,10 @@
 #ifndef VIEWSTYLE_H
 #define VIEWSTYLE_H
 
+// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+#include "Scintilla.h"
+// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
+
 namespace Scintilla::Internal {
 
 /**
@@ -67,6 +71,14 @@ struct CaretLineAppearance {
 	int frame = 0;
 };
 
+// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+constexpr int GetFontSizeZoomed(int size, int zoomLevel) noexcept {
+	size = (size * zoomLevel + 50) / 100;
+	// Hangs if sizeZoomed (in point) <= 1
+	return std::max(size, 2 * FontSizeMultiplier);
+}
+// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
+
 struct CaretAppearance {
 	// Line, block, over-strike bar ...
 	Scintilla::CaretStyle style = CaretStyle::Line;
@@ -148,7 +160,9 @@ public:
 	int fixedColumnWidth = 0;	///< Total width of margins
 	bool marginInside;	///< true: margin included in text view, false: separate views
 	int textStart;	///< Starting x position of text within the view
-	int zoomLevel;
+	// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+	int zoomLevel;  /// @ 2018-09-06 Changed to a percent value
+	// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 	Scintilla::WhiteSpace viewWhitespace;
 	Scintilla::TabDrawMode tabDrawMode;
 	int whitespaceSize;
@@ -245,11 +259,15 @@ public:
 	bool IsCaretVisible(bool isMainSelection) const noexcept;
 	bool DrawCaretInsideSelection(bool inOverstrike, bool imeCaretBlockOverride) const noexcept;
 	CaretShape CaretShapeForMode(bool inOverstrike, bool isMainSelection) const noexcept;
+	// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+	bool ZoomIn() noexcept;
+	bool ZoomOut() noexcept;
+	// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 
 private:
 	void AllocStyles(size_t sizeNew);
 	void CreateAndAddFont(const FontSpecification &fs);
-	FontRealised *Find(const FontSpecification &fs);
+	FontRealised *Find(const FontSpecification &fs) const;
 	void FindMaxAscentDescent() noexcept;
 };
 
