diff --git a/_upgrade_staging/scintilla/src/CellBuffer.cxx b/scintilla/src/CellBuffer.cxx
index 3e9deb934..165dccedd 100644
--- a/_upgrade_staging/scintilla/src/CellBuffer.cxx
+++ b/scintilla/src/CellBuffer.cxx
@@ -21,12 +21,14 @@
 #include <optional>
 #include <algorithm>
 #include <memory>
-#include <type_traits>
 
 #include "ScintillaTypes.h"
 
 #include "Debugging.h"
 
+// >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+#include "Scintilla.h"
+// <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 #include "Position.h"
 #include "SplitVector.h"
 #include "Partitioning.h"
@@ -216,7 +218,7 @@ public:
 	}
 	void InsertLines(Sci::Line line, const Sci::Position *positions, size_t lines, bool lineStart) override {
 		const POS lineAsPos = pos_cast(line);
-		if constexpr (std::is_convertible_v<Sci::Position *, POS *>) {
+		if constexpr (sizeof(Sci::Position) == sizeof(POS)) {
 			starts.InsertPartitions(lineAsPos, positions, lines);
 		} else {
 			starts.InsertPartitionsWithCast(lineAsPos, positions, lines);
