diff --git a/_upgrade_staging/lexilla/lexlib/StyleContext.h b/lexilla/lexlib/StyleContext.h
index d148140e8..f188d1bef 100644
--- a/_upgrade_staging/lexilla/lexlib/StyleContext.h
+++ b/lexilla/lexlib/StyleContext.h
@@ -187,6 +187,24 @@ public:
 	void GetCurrentLowered(char *s, Sci_PositionU len) const;
 	enum class Transform { none, lower };
 	void GetCurrentString(std::string &string, Transform transform) const;
+
+    // >>>>>>>>>>>>>>>   BEG NON STD SCI PATCH   >>>>>>>>>>>>>>>
+
+	bool Match(char ch0, char ch1, char ch2) const noexcept {
+		return Match(ch0, ch1) && ch2 == styler.SafeGetCharAt(currentPos + 2, '\0');
+	}
+
+	bool MatchNext() const noexcept {
+		return ch == chNext && ch == static_cast<unsigned char>(styler[currentPos + 2]);
+	}
+	bool MatchNext(char ch0, char ch1) const noexcept {
+		return chNext == static_cast<unsigned char>(ch0) && ch1 == styler.SafeGetCharAt(currentPos + 2, '\0');
+	}
+    bool MatchNext(char ch0, char ch1, char ch2) const noexcept {
+        return MatchNext(ch0, ch1) && ch2 == styler.SafeGetCharAt(currentPos + 3, '\0');
+    }
+
+    // <<<<<<<<<<<<<<<   END NON STD SCI PATCH   <<<<<<<<<<<<<<<
 };
 
 }
